internal/kibana/route.go

```diff
@@ -0,0 +1,189 @@
+	if err != nil {
+		return kverrors.Wrap(err, "failed to update Kibana route for cluster",
+			"cluster", cluster.Name,
+			"namespace", cluster.Namespace,
+		)
+	}
+
+	log.V(1).Info(fmt.Sprintf("Successfully reconciled kibana route: %s", res),
+		"route_name", rt.Name,
+		"cluster", cluster.Name,
+		"namespace", cluster.Namespace,
+	)
+
+	return nil
+}
+
+func (clusterRequest *KibanaRequest) createOrUpdateKibanaConsoleLink() error {
+	cluster := clusterRequest.cluster
+
+	kibanaURL, err := clusterRequest.GetRouteURL("kibana")
+	if err != nil {
+		return kverrors.Wrap(err, "failed to get route URL for kibana")
+	}
+
+	cl := console.NewConsoleLink(KibanaConsoleLinkName, kibanaURL, "Logging", "Observability")
+
+	res, err := console.CreateOrUpdateConsoleLink(context.TODO(), clusterRequest.client, cl, console.CompareConsoleLinks, console.MutateConsoleLinkSpecOnly)
+	if err != nil {
+		return kverrors.Wrap(err, "failed to create or update kibana console link CR for cluster",
+			"cluster", cluster.Name,
+		)
+	}
+
+	log.V(1).Info(fmt.Sprintf("Successfully reconciled kibana consolelink: %s", res),
+		"console_link_name", cl.Name,
+		"cluster", cluster.Name,
+	)
+
+	return nil
+}
+
+func (clusterRequest *KibanaRequest) createOrUpdateKibanaConsoleExternalLogLink() (err error) {
+	cluster := clusterRequest.cluster
+
+	kibanaURL, err := clusterRequest.GetRouteURL("kibana")
+	if err != nil {
+		return kverrors.Wrap(err, "failed to get route URL", "cluster", clusterRequest.cluster.Name)
+	}
+
+	labels := map[string]string{
+		"component":     "support",
+		"logging-infra": "support",
+		"provider":      "openshift",
+	}
+
+	consoleExternalLogLink := console.NewConsoleExternalLogLink(
+		"kibana",
+		"Show in Kibana",
+		strings.Join([]string{
+			kibanaURL,
+			"/app/kibana#/discover?_g=(time:(from:now-1w,mode:relative,to:now))&_a=(columns:!(kubernetes.container_name,message),query:(query_string:(analyze_wildcard:!t,query:'",
+			strings.Join([]string{
+				"kubernetes.pod_name:\"${resourceName}\"",
+				"kubernetes.namespace_name:\"${resourceNamespace}\"",
+				"kubernetes.container_name.raw:\"${containerName}\"",
+			}, " AND "),
+			"')),sort:!('@timestamp',desc))",
+		},
+			""),
+		labels,
+	)
+
+	res, err := console.CreateOrUpdateConsoleExternalLogLink(
+		context.TODO(),
+		clusterRequest.client,
+		consoleExternalLogLink,
+		console.CompareConsoleExternalLogLinkEqual,
+		console.MutateConsoleExternalLogLink,
+	)
+	if err != nil {
+		return kverrors.Wrap(err, "failed to create or update kibana console external log link CR for cluster",
+			"cluster", cluster.Name,
+			"kibana_url", kibanaURL,
+		)
+	}
+
+	log.V(1).Info(fmt.Sprintf("Successfully reconciled kibana external log link: %s", res),
+		"console_external_log_link_name", consoleExternalLogLink.Name,
+		"cluster", cluster.Name,
+	)
+
+	return nil
+}
+
+func (clusterRequest *KibanaRequest) removeSharedConfigMapPre45x() error {
```

Comment:
we can drop this function already... we require being on 4.6 before you can go to 5.x

Reply:
Done